runtime/libc/wasi_sdk_backing.c

```diff
@@ -27,21 +27,63 @@
 #define fdatasync fsync
 #endif
 
-int main(int argc, char* argv[]) {
-    runtime_main(argc, argv);
-    printf("mem use = %d\n", (int) memory_size);
+// Code that actually runs the wasm code
+IMPORT void wasmf__start(void);
+u32 runtime_argc = 0;
+char *runtime_argv_buffer = NULL;
+u32 runtime_argv_buffer_len = 0;
+u32 *runtime_argv_buffer_offsets = NULL;
+
+void runtime_cleanup() {
+    free(runtime_argv_buffer);
+    runtime_argv_buffer_len = 0;
+    free(runtime_argv_buffer_offsets);
 }
 
-// The symbol the binary gives us to init libc
-extern void wasmf___wasm_call_ctors(void);
+int main(int argc, char* argv[]) {
+    // Set argc and argv to globals, these are later used by the WASI syscalls 
+    runtime_argc = argc;
+
+    runtime_argv_buffer_offsets = calloc(argc, sizeof(u32));
```

Comment:
Use a global variable, not calloc. Don't want to allocate in the runtime generally

Reply:
I don't think that's possible here as this is allocating the array of arguments, and we don't statically know how many arguments there are.